### PR TITLE
Improve documentation for additional rules

### PR DESCRIPTION
**Pull request by daniel-beck**
#3 continued.

### CODE REVIEW

File: jenkins/java/ql/src/FileFromRestOfPath.md [hunk 1]
```diff
@@ -1 +1,28 @@
...
+2. Determine whether this finding is a false positive (see guidance below). This is an automated scan result, so that's always a possibility. In general, the rules err on the side of caution, so false positives are relatively common. If it is a false positive, mark it as such on the GitHub UI and you're done!
+3. If you have determined that this is a true positive finding, review the documentation below how to resolve it.
+
+If this finding is in a plugin hosted by the Jenkins project, you can also always [contact the Jenkins Security Team via Jira or email](https://www.jenkins.io/security/#reporting-vulnerabilities) to ask for help in resolving this finding.
+
+## Is this a false positive?
+
+It's legitimate to use `#getRestOfPath` to access a file. `DirectoryBrowserSupport` and other classses in Jenkins (core) do that as well. However, the value needs to be carefully sanitized, or the code needs to reject any values that could be used for path traversal. If your code is already doing that, and this check just didn't identify that, great!
```
Comment (jmMeessen) [inline jenkins/java/ql/src/FileFromRestOfPath.md:19]: typo in `classses`

File: jenkins/java/ql/src/FileFromRestOfPath.md [hunk 2]
```diff
@@ -1 +1,28 @@
-`#getRestOfPath` and/or the file path resulting from its use needs to be carefully sanitized to not result in a path traversal vulnerability.
+The value from `#getRestOfPath` is used to build a `File`, resulting in a potential path traversal vulnerability.
+
+## Why is this a problem?
+
+If your code is modifying or reading a file as specified in the URL, users able to access this URL may be able to access files outside your intended root directory unless you're restricting allowed values.
```
Comment (jmMeessen) [inline jenkins/java/ql/src/FileFromRestOfPath.md:5]: Phrase is a trifle too long. But can't propose anything better. Thought of putting "able to access this URL" between commas. But this doesn't work well with the first segment.

So this is good to go.

File: jenkins/java/ql/src/HasPermissionReturnIgnored.md [hunk 3]
```diff
@@ -1,7 +1,25 @@
-== Ignoring return value of `#hasPermission` calls
-
 This code calls `#hasPermission` without checking the return value. This call should either be `#checkPermission`, or have its return value used.
 
+## Why is this a problem?
+
+This is indicative of a bug: Code intended to ensure the user accessing some functionality has a given permission, but ignoring the return value of this method means that anyone would be able to proceed.
```
Comment (jmMeessen) [inline jenkins/java/ql/src/HasPermissionReturnIgnored.md:5]: I would suggest shorter phrases. How about: 

"This is indicative of a bug. The code  wants to ensure that, to access a functionality, the user  holds a given permission. But the method's return value is ignored. This means that anyone can access the functionality."

**Review by jmMeessen — COMMENTED** [review]
LGTM
Just made some style comments or suggestion

**Review by jmMeessen — APPROVED** [review]

File: jenkins/java/ql/src/FileFromRestOfPath.md (hunk 1 above)
Comment (daniel-beck) [reply jenkins/java/ql/src/FileFromRestOfPath.md:19]:
```suggestion
It's legitimate to use `#getRestOfPath` to access a file. `DirectoryBrowserSupport` and other classes in Jenkins (core) do that as well. However, the value needs to be carefully sanitized, or the code needs to reject any values that could be used for path traversal. If your code is already doing that, and this check just didn't identify that, great!
```

**Pull request merged**